dir.props

```diff
@@ -4,6 +4,11 @@
 
   <Import Project="dir.common.props" />
 
+  <PropertyGroup>
+    <LangVersion>8.0</LangVersion>
+    <UseSharedCompilation>true</UseSharedCompilation>
```

Comment:
Pushed a commit setting `UseSharedCompilation=true`, as we did on CoreFX.

When using a non-toolset compiler this is the default, but for Toolset builds the reverse is true and this is disabled. This should speed up the build and avoid the timeout that was getting hit.